src/remote/action.go

```diff
@@ -73,12 +73,24 @@ func (c *Client) buildAction(target *core.BuildTarget, isTest, stamp bool) (*pb.
 	return command, actionDigest, nil
 }
 
+// stateForTarget returns an appropriate state for the current target.
+// TODO(peterebden): This is very much a limitation of the current setup; there is a complex interaction between how we set
+//                   HOME and how we get the correct state object for cross-compiling.
+//                   When we release v16 and make this the default we should be able to significantly simplify this.
+func (c *Client) stateForTarget(target *core.BuildTarget) *core.BuildState {
+	if !c.state.Config.FeatureFlags.PleaseDownloadTools {
+		return c.state
+	}
+	return c.state.ForTarget(target)
```

Comment:
Can't we just do:
```
s := c.state.ForTarget(target)
s.Config.HomeDir = s.Config.Remote.HomeDir
return s
```

Reply:
No, because we don't want to change it for anything that executes locally using that same state object.